core/LSU.cpp

```diff
@@ -1,9 +1,14 @@
-
 #include "sparta/utils/SpartaAssert.hpp"
 #include "LSU.hpp"
 
 namespace olympia
 {
+    inline core_types::RegFile determineRegisterFile(const mavis::OperandInfo::Element & reg)
+    {
+        const bool is_float = (reg.operand_type == mavis::InstMetaData::OperandTypes::SINGLE) ||
+                              (reg.operand_type == mavis::InstMetaData::OperandTypes::DOUBLE);
+        return is_float ? core_types::RegFile::RF_FLOAT : core_types::RegFile::RF_INTEGER;
```

Comment:
This is great, but to flag an early error to future developers, consider a full check.  And maybe create a map for speed:
```suggestion
        static const std::map<mavis::InstMetaData::OperandTypes, core_types::RegFile> mavis_optype_to_regfile = { 
            // mapping of supported types ... 
         };

        if(auto match = mavis_optype_to_regfile.find(reg); match != mavis_optype_to_regfile.end()) {
            return match->second;
        }
        sparta_assert(false, "Unknown reg type: " << ... more info);
        return core_types::RegFile::UNKNOWN;

```

Reply:
Addressed in [4fd928d](https://github.com/riscv-software-src/riscv-perf-model/pull/51/commits/4fd928d4400af49de00ac3f5f8b5c4990bdbb2fd)